src/ILogger/Readme.md

```diff
@@ -0,0 +1,195 @@
+    {
+        public static void Main(string[] args)
+        {
+            var host = BuildWebHost(args);
+            var logger = host.Services.GetRequiredService<ILogger<Program>>();
+            logger.LogInformation("From Program. Running the host now.."); // This will be picked up up by AI
+            host.Run();
+        }
+
+        public static IWebHost BuildWebHost(string[] args) =>
+            WebHost.CreateDefaultBuilder(args)
+                .UseStartup<Startup>()                
+                .ConfigureLogging(logging =>
+                {
+                    logging.ClearProviders(); //optionally removing all other logging providers.
+                    logging.AddApplicationInsights("ikeyhere");
+					// Optional: Apply filters to configure LogLevel Trace or above is sent to ApplicationInsights for all
+					// categories.
+					logging.AddFilter<ApplicationInsightsLoggerProvider>("", LogLevel.Trace);
+					// Additional filtering For category starting in "Microsoft",
+					// only Warning or above will be sent to Application Insights.
+                    logging.AddFilter<ApplicationInsightsLoggerProvider>("Microsoft", LogLevel.Warning);
+                })
+                .Build();
+    }
+```
+
+```
+public class Startup
+    {
+        private readonly ILogger _logger;
+
+        public Startup(IConfiguration configuration, ILogger<Startup> logger)
+        {
+            Configuration = configuration;
+            _logger = logger;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);                         
+             _logger.LogInformation("From ConfigureServices. Services.AddMVC invoked"); // This will be picked up up by AI
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                _logger.LogInformation("Configuring for Development environment");
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                _logger.LogInformation("Configuring for Production environment");
+            }
+
+            app.UseMvc();
+        }
+    }
+```
+
+```
+public class ValuesController : ControllerBase
+    {
+        private readonly ILogger _logger;
+
+        public ValuesController(ILogger<ValuesController> logger)
+        {
+            _logger = logger;
+        }
+
+        // GET api/values
+        [HttpGet]
+        public ActionResult<IEnumerable<string>> Get()
+        {
+            _logger.LogInformation("This is an information trace..");
```

Comment:
Might be worth adding a small sample dealing with using Scopes. ILogger implementation particularly likes Dictionary of string to objects and exposes them in custom dimensions. Something that people might find useful.

Reply:
This can be added here and above in console too.